### ISSUE TITLE
Object keys containing spaces not supported

### ISSUE THREAD

**Issue by frxnz** [rexm/Handlebars.Net#126]
I have a Dictionary whose keys contain spaces. Something like `{ "Dining Type", "Restaurant" }`. In the JS version of Handlebars you would reference this like `{{Entries.[Dining Type]}}`.

Using that syntax I get the following error:

```
{"Template references a helper that is not registered. Could not find helper 'Entries.[Dining'"}
```

Using the alternate syntax (`{{Entries."Dining Type"}}`) I don't get an error but no value is rendered.

Is this feature not supported or am I doing something wrong?

**Comment by frxnz** [comment]
There also appears to be a bug in the Dictionary support in general. `{{ [expression] }}` works but `{{[expression]}}` doesn't.

**Comment by rexm** [comment]
Definitely an issue with the lexer. Not sure when I'll have time to get to this; if it's urgent, PR always welcome!

**Comment by frxnz** [comment]
PR submitted #128 :+1:

**Comment by rexm** [comment]
Partially fixed in #128 & released in nuget 1.7.1. `name.[with space]` now supported, but `name."with space"` still not.

**Comment by imsedim** [comment]
Something is broken after #128. 

This

    Hello, {{link_to names.[1].name \"title\"}}!


Works in version 1.7.0 (https://dotnetfiddle.net/xw5sDE)
Does not work in version 1.7.1 (https://dotnetfiddle.net/lhI5pE)

**Comment by rexm** [comment]
This was addressed in 1.8.0
